pkg/k8scontext/context.go

```diff
@@ -347,6 +351,27 @@ func (c *Context) GetService(serviceKey string) *v1.Service {
 	return service
 }
 
+// GetPodsByServiceSelector returns pods that are associated with a specific service.
+func (c *Context) GetPodsByServiceSelector(selector map[string]string) []*v1.Pod {
+	podList := make([](*v1.Pod), 0)
+	podListInterface := c.Caches.Pods.List()
+	for _, podInterface := range podListInterface {
+		pod := podInterface.(*v1.Pod)
+		match := true
+		for key, item := range selector {
```

Comment:
Feel free to ignore this - just lamenting the lack of built-in Set primitives in go.
---
This is very interesting.  It sounds like you have 2 sets.  One is the pod labels, the ohter the selector keys.
It would have been elegant to do an Intersect of the the 2 sets and see if there's an overlap. That'd be a clear indicator that we have a match.

Our own `utils.UnorderedSet` library implements an `Intersect`: https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/pkg/utils/unorderedset.go#L83-L91 
Using that seems tricky though - there are sideffects - it does a mutation of the set passet to Intersect (ouch).

Much better implementation is offered here: https://github.com/deckarep/golang-set/blob/504e848d77ea4752b3057b8fb46da0e7f746ccf3/set.go#L83-L90
(I hope we get a chance to swap our implementation with that library)

Here is their implementation: https://github.com/deckarep/golang-set/blob/699df6a3acf6867538e50931511e9dc403da108a/threadunsafe.go#L116-L135